메인페이지를 구성한다

GyeongyeonAn opened the pull request:
메인페이지 구성을 완료했습니다.

style: 적절한 위치 배정 완료
chore: 링크 이동 구현 완료

![image](https://user-images.githubusercontent.com/55589616/118397248-15115200-b68e-11eb-8181-62969965c4d4.png)

체크 부탁드립니다.

close #10

GyeongyeonAn committed:
chore: add text

GyeongyeonAn committed:
Merge branch 'master' of https://github.com/DonggukBOB/dongguk-bob into feature/10

GyeongyeonAn committed:
chore: image 추가

GyeongyeonAn committed:
Merge branch 'feature/11' of https://github.com/DonggukBOB/dongguk-bob into feature/10

GyeongyeonAn committed:
Merge branch 'master' of https://github.com/DonggukBOB/dongguk-bob into feature/10

GyeongyeonAn committed:
Merge branch 'master' of https://github.com/DonggukBOB/dongguk-bob into feature/10

GyeongyeonAn committed:
style: 적절한 위치 지정  - 컴포넌트 크기 및 위치 조정  chore: 버튼에 링크 이동 구현

hwookim commented on src/pages/main/index.jsx:
```diff
@@ -1,5 +1,62 @@
-  return <div>Main</div>;
+  return (
+    <>
+      <Header />
```
메인페이지에 헤더가 필요한가요?
논의 때는 없기로 했던 거 같아요.

GyeongyeonAn replied:
네 헤더 없애는게 맞아요! 수정할게요.

hwookim commented on src/pages/main/index.jsx:
```diff
@@ -1,5 +1,62 @@
+        <SubSubTitle>든-든한</SubSubTitle>
+        <Title>동국밥</Title>
+        <Image src={imgMain} />
+        <Link to="../recommend/index.jsx">
+          <Button>메뉴 추천</Button>
+        </Link>
```
```suggestion
        <Link to="/recommend">
          <Button>메뉴 추천</Button>
        </Link>
```
Link에는 그냥 url만 쓰면 됩니다.

GyeongyeonAn replied:
넵 수정할게요!

hwookim commented on src/pages/main/index.jsx:
```diff
@@ -1,5 +1,62 @@
 import React from "react";
+import imgMain from "./imgMain.jpg";
```
이미지는 src폴더 밖에 별도의 assets 폴더를 만들어서 관리하는 건 어때요?
코드 구획에는 코드만 담는 것은 어떨까요?

GyeongyeonAn replied:
네 src폴더 밖에 assets 폴더를 만들겠습니다.

hwookim commented on src/pages/main/index.jsx:
```diff
@@ -1,5 +1,62 @@
+  font-size: 40pt;
+`;
+
+const SubTitle = styled.h3`
+  text-align: center;
+  margin-top: 30px;
+`;
+
+const SubSubTitle = styled.h3`
+  text-align: center;
+`;
```
상위 컨테이너에 text-align이 되어있다면 각 객체마다 걸어 줄 필요는 없습니다.
필요없는 중복 코드는 삭제해주세요.

GyeongyeonAn replied:
네 삭제할게요!

hwookim commented on src/pages/main/index.jsx:
```diff
@@ -1,5 +1,62 @@
+import Header from "../../components/Header";
+import { Link } from "react-router-dom";
+
+const Container = styled.div`
+  text-align: center;
+`;
```
컨테이너는 통상적으로 해당 컴포넌트 전체를 감쌀 때 사용합니다.
최상위를 <></>로 감싸지 말고 Container로 감싸는 것이 더 좋아보이네요.

SubTitle에 margin을 주기보단 Container에서 padding을 통해 관리하는 방법도 있겠네요.

GyeongyeonAn replied:
아 그렇네요 오케이 padding으로 하겠습니다

hwookim replied:
근데 padding으로 해도 브라우저 사이즈별 대책이 제대로 세워지진 않을 것 같아요.
세로 중앙 정렬을 하면 더 좋지 않을까요?

GyeongyeonAn replied:
제대로 이해를 못했어요
컨테이너에 아래 코드를 넣으면 되는 건가요?  

  flex-direction: column;
  align-items: center;
  justify-content: center;

hwookim commented on src/pages/main/index.jsx:
```diff
@@ -1,5 +1,62 @@
+
+const Title = styled.h1`
+  letter-spacing: 10px;
+  font-size: 40pt;
```
이친구는 갑자기 왜 pt...?
단위는 통일하는 게 좋을 거 같아요

GyeongyeonAn replied:
네 수정했습니다.

hwookim commented on src/pages/main/index.jsx:
```diff
@@ -1,5 +1,62 @@
+`;
+
+const Image = styled.img`
+  width: 400px;
+  height: 400px;
```
px 단위보단 %, em, rem 같은 가변 단위를 쓰는 것이 반응형 웹에 더 유연합니다.
실제로 현재 모바일 화면 상에는 버튼이 튕겨나가네요.

![스크린샷 2021-05-17 오후 10 35 46](https://user-images.githubusercontent.com/45786387/118497651-3b072700-b760-11eb-9582-4bde210f9819.png)

hwookim commented on src/pages/main/index.jsx:
```diff
@@ -1,5 +1,62 @@
+  width: 400px;
+  height: 400px;
+  display: block;
+  margin-left: auto;
+  margin-right: auto;
```
```suggestion
  margin: auto;
```
이렇게 쓸 수도 있어요.

hwookim commented on src/pages/main/index.jsx:
```diff
@@ -1,5 +1,62 @@
+const Image = styled.img`
+  width: 400px;
+  height: 400px;
+  display: block;
```
넣은 이유가 있나요?

hwookim commented on src/pages/main/index.jsx:
```diff
@@ -1,5 +1,62 @@
+  border: none;
+  font-size: 25px;
+  margin: 10px;
+  color: #fff;
```
```suggestion
  color: white;
```
이렇게 쓸 수도 있슴당

hwookim reviewed (changes requested):
버튼을 조금 작게 하는 건 어떠려나요?

![스크린샷 2021-05-17 오후 10 53 54](https://user-images.githubusercontent.com/45786387/118500466-c41f5d80-b762-11eb-8400-6a7e90637066.png)

만드실 때 개발자도구에서 모바일 화면도 한 번쯤은 확인해보시면 좋을 것 같아요.

container에서 flex를 통해 가운데 중앙 정렬을 해보는 건 어떤가요?
margin으로 위의 여백을 처리하다보면 기기마다 위치가 이상하게 보일 수 있어요.
css를 통해 배치하기 힘드시면 제가 커밋을 올려도 괜찮구요.

GyeongyeonAn committed:
style: css 수정
- flex를 이용한 가로, 세로 가운데 정렬
- 폰트 사이즈 em 활용하는 것으로 변경
- 전체를 Container로 묶음

hwookim commented on src/pages/main/index.jsx:
```diff
@@ -1,62 +1,55 @@
-
-const SubSubTitle = styled.h3`
-  text-align: center;
+  font-size: 1em;
```
1em이면 그냥 안써도 됩니다.

GyeongyeonAn replied:
아 디폴트가 1em이군요! 이해했습니다

hwookim commented on src/pages/main/index.jsx:
```diff
@@ -1,62 +1,55 @@
+  align-items: center;
+  justify-content: center;
+  padding: 10%;
 `;
```
font-size를 부모에서 지정해버리면 하위 컴포넌트들이 em을 쓸 이유가 없어집니다.
em은 부모컴포넌트의 px값을 기반으로 진행되거든요.

justify-content를 통해서 세로 중앙 정렬을 하고싶으시다면 height를 100%로 설정해야합니다.
현재로서는 의미 없는 코드가 되었고, padding을 통해 간격을 벌리게 되었네요 ㅠ

align-items 쓰시면 text-align은 안써도 되긴 합니다.

GyeongyeonAn replied:
컨테이너 안에

  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

다음과 같이 수정합니다.

hwookim commented on src/pages/main/index.jsx:
```diff
@@ -1,62 +1,55 @@
-  margin: 10px;
-  color: #fff;
+  font-size: 1.5em;
+  margin: 3%;
```
버튼 사이의 간격을 조절하는 거라면 일정한 px 단위가 더 어울릴 거 같아요.

GyeongyeonAn replied:
네
  margin: 5px;
로 수정합니다.

hwookim commented on src/pages/main/index.jsx:
```diff
@@ -1,62 +1,55 @@
-  margin-right: auto;
+  width: 100%;
+  height: 100%;
+  margin: auto;
```
margin: auto의 역할은 무엇인가요?

GyeongyeonAn replied:
디자인을 검색해서 하다보니 실수가 있습니다. 삭제해도 변화가 없길래 삭제했습니다!

hwookim commented on src/pages/main/index.jsx:
```diff
@@ -1,62 +1,55 @@
+      <SubTitle>든-든한</SubTitle>
+      <Title>동국밥</Title>
+      <Image src={imgMain} />
+      <Link to="/recommend">
+        <Button>메뉴 추천</Button>
+      </Link>
+      <Link to="/list">
+        <Button>맛집 목록</Button>
+      </Link>
```
현재 flex-column 상태이기 때문에 감싸는 컨테이너가 하나 더 있어야 원래처럼 가로로 정렬할 수 있게 됩니다.

현재는 아래처럼 나오네요.
![스크린샷 2021-05-20 오전 1 56 02](https://user-images.githubusercontent.com/45786387/118853276-89f2bf00-b90e-11eb-94b7-8bf862f2ea1a.png)

white-space: nowrap; 을 통해 글자의 줄바꿈을 제한할 수 있습니다.
이후에 크기 조절은 padding을 통해서 진행할 수 있고요.

hwookim reviewed (changes requested):
화이팅

GyeongyeonAn committed:
refactor: 코드 수정
- Container에 불필요한 font-size, padding, text-align 삭제
- Container에 height: 100% 추가
- 버튼 사이의 간격 margin 10px로 변경
- Image에 불필요한 margin: auto 삭제
- 버튼에 white-space: nowrap 추가
- 버튼에 padding으로 크기 조정
- 버튼에 버튼랩으로 감싸서 가로로 정렬
- Link to 삭제하고 useHistory와 onClick 사용하는 방법으로 변경

hwookim commented on src/pages/main/index.jsx:
```diff
@@ -1,5 +1,59 @@
+
+const Image = styled.img`
+  width: 100%;
+  height: 55%;
```
이미지같은 경우에는 width에 따라서 height가 자동적으로 할당되니 괜히 안건드리는 편이 더 좋을 수 있습니다.
따로 지정하게 되면 각각을 맞추느라 사진이 찌그러질 수 있어요.
이 경우에는 얼추 비율이 맞아 괜찮지만, 알고계시면 좋을 듯 합니다.

GyeongyeonAn replied:
아하 그렇군요 그럼 height는 지우고 확인해보고 괜찮으면 반영해서 머지할게요

hwookim reviewed (approved):
고생하셨습니다.
하나 추가적으로 리뷰 남겼으니 확인해주시고 머지해주세요

GyeongyeonAn committed:
refactor: 이미지 크기 조정
- Image에 height 삭제 (크기가 찌그러질 수 있는 문제 때문에)
